tasks/adventure_solutions/task_7_while.py

```diff
@@ -0,0 +1,81 @@
+import random
+import time
+
+def wrong_input(valid):
+    print(f"Sorry but the only valid options are: {valid}.")
+
+def dead(death_message):
+    print(f"Sorry {name}, {death_message}.")
+    print("You're falling for...")
+
+    for number in range(1,11):
+        print(number)
+        time.sleep(1)
+
+    print("meters!")
+    print("You are dead!")
+
+friends = ['Emily', 'Aurora', 'Jack', 'Daniel', 'Meghan']
+
+door_greetings = {'1': "Welcome to paradise!",
+                  '2': "Welcome to the dormitory!"}
+
+print()
+name = input("What is your name? ")
+print(f"Welcome to the dungeon, {name}!")
+
+print("Do you go through door 1 or door 2?")
+door = input("> ")
+
+while door != "1" and door != "2":
+    wrong_input("1, 2")
+    print("Let's try again: do you go through door 1 or door 2?")
+    door = input("> ")
+
+
+if door == "1":
+    print(f"{door_greetings[door]}")
+    print(f"Look who's here! Your friend {friends[random.randint(0,4)]} :D")
+
+    print("There is a nice vampire asking you if you enjoy life.")
+    print("What do you do?")
+    print("A. Smile and nod")
+    print("B. Scream and run")
+
+    vampire = input("> ")
+    while vampire != "A" and vampire != "B":
```

Comment:
I wonder if it would be nice to show here an alternative way to check this eg. `vampire in {'A', 'B'}` which is if this element belongs to this set. It is a bit more elegant imho. Just an idea, it is your call

Reply:
oh, I was meaning to search a better way of doing it. didn't know about the "in".